sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpProducer.cs

```diff
@@ -203,8 +203,27 @@ internal class AmqpProducer : TransportProducer
             Argument.AssertNotClosed(_closed, nameof(AmqpProducer));
             Argument.AssertNotClosed(ConnectionScope.IsDisposed, nameof(EventHubConnection));
 
-            AmqpMessage messageFactory() => MessageConverter.CreateBatchFromEvents(events, sendOptions?.PartitionKey);
```

Comment:
Yehaa. This factory was ugly... :partying_face:

Reply:
yeah, it was.  The pivot to "keep events because the AMQP messages are throwing on retries" was very late in the ship cycle.